src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -158,9 +154,23 @@ def validate_filter_parameter(string):
     result = ()
     if string:
         comps = string.split('=', 1)
+
         # Ignore invalid arguments like  '=value' or '='
         if comps[0]:
-            result = (comps[0], comps[1]) if len(comps) > 1 else (string, '')
+            if len(comps) > 1:
+                # In the portal, if value textbox is blank we store the value as empty string.
+                # In CLI, we should allow inputs like 'name=', which correspond to empty string value.
+                # But there is no way to differentiate between CLI inputs 'name=' and 'name=""'.
+                # So even though "" is invalid JSON escaped string, we will accept it and set the value as empty string.
+                filter_param_value = '\"\"' if comps[1] == "" else comps[1]
+                try:
+                    # Ensure that provided value of this filter parameter is valid JSON. Error out if value is invalid JSON.
+                    filter_param_value = json.loads(filter_param_value)
+                except ValueError:
+                    raise CLIError('Filter parameter value must be a JSON escaped string. "{}" is not a valid JSON object.'.format(filter_param_value))
+                result = (comps[0], filter_param_value)
+            else:
+                result = (string, '')
         else:
             logger.warning("Ignoring filter parameter '%s' because parameter name is empty.", string)
```

Comment:
We can consider this is invalid if it is not in "key=value" format (preferred).  So no "=" or "=value" will fail explicitly. 
If you want to have some tolerance here, we should add warning log for both case.

Reply:
Okay, I think it makes sense to error out on `=` or `=value` input since portal also does the same.

Reply:
If the input format is not key=value, in line 172, we can also throw exception instead of treat input as key and value is empty string. 

---
In reply to: [558463404](https://github.com/Azure/azure-cli/pull/16536#discussion_r558463404) [](ancestors = 558463404)

Reply:
If line 172 condition is met, it means that the input must me something like `--filter-parameters name`. This is something we can support and treat the value as empty string (same behavior as portal).